service/controller/resource/cloudconfigblob/create.go

```diff
@@ -27,6 +27,11 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	if !azureMachinePool.GetDeletionTimestamp().IsZero() {
```

Comment:
Do we need the same check for `MachinePool` CR here?

Reply:
Is it even possible that `EnsureCreated` is called for an `AzureMachinePool` CR that is being deleted?

Reply:
I think so. IIUC controllers get the full list of items every X minutes. That may include deleted items? I can't confirm 100% though.

Reply:
No. `EnsureCreated()` won't be called on object that is deleted (i.e. has non-zero `deletionTimestamp`).

https://github.com/giantswarm/operatorkit/blob/master/pkg/controller/controller.go#L461-L489

Reply:
I removed the check for the reconciled CR and added checks for other CRs that are used but are not the ones being watched.